actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -438,7 +449,20 @@ def form_authenticity_param # :doc:
 
       # Checks if the controller allows forgery protection.
       def protect_against_forgery? # :doc:
-        allow_forgery_protection
+        allow_forgery_protection && ensure_session_is_enabled!
+      end
+
+      def ensure_session_is_enabled!
+        if !session.respond_to?(:enabled?) || session.enabled?
+          true
+        else
+          if Base.silence_disabled_session_errors
+            ActiveSupport::Deprecation.warn(ActionController::RequestForgeryProtection::DisabledSessionError::MESSAGE)
+            false
+          else
+            raise ActionController::RequestForgeryProtection::DisabledSessionError
+          end
```

Comment:
Do we need to use the fully qualified constant in this block? Will not `DisabledSessionError::MESSAGE` and `DisabledSessionError` work?

Reply:
No indeed, that's a leftover from before I moved this code here. Corrected.